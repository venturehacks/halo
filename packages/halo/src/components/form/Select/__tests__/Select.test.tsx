import { render } from 'enzyme';
import React from 'react';

import { Select, SelectOption } from '../index';

const options: SelectOption[] = [
  { label: 'Chocolate', value: 'chocolate' },
  { label: 'Vanilla', value: 'vanilla' },
  { label: 'Strawberry', value: 'strawberry' },
];

describe('Select', () => {
  test('smoke', () => {
    const component = render(<Select options={options} />);

    expect(component.text()).toContain('ChocolateVanillaStrawberry');
  });

  describe('snapshot', () => {
    describe('size', () => {
      test('sm', () => {
        const component = render(<Select options={options} size="sm" />);
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(<Select options={options} />);
        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = render(<Select options={options} size="lg" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
