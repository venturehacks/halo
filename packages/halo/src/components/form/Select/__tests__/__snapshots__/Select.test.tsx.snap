// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot size lg 1`] = `
<select
  class="text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 p-3 h-12 border border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-light-aa w-auto component"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;

exports[`Select snapshot size md 1`] = `
<select
  class="text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 px-3 py-2 h-10 border border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-light-aa w-auto component"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;

exports[`Select snapshot size sm 1`] = `
<select
  class="text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 px-3 py-1 h-8 border border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-light-aa w-auto component"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;
